httpserver/src/main/kotlin/tbdex/sdk/httpserver/handlers/SubmitClose.kt

```diff
@@ -0,0 +1,73 @@
+package tbdex.sdk.httpserver.handlers
+
+import io.ktor.http.*
+import io.ktor.server.application.*
+import io.ktor.server.request.*
+import io.ktor.server.response.*
+import tbdex.sdk.httpclient.models.ErrorDetail
+import tbdex.sdk.httpserver.models.CallbackError
+import tbdex.sdk.httpserver.models.ErrorResponse
+import tbdex.sdk.httpserver.models.ExchangesApi
+import tbdex.sdk.httpserver.models.SubmitCallback
+import tbdex.sdk.protocol.models.Close
+import tbdex.sdk.protocol.models.Message
+import tbdex.sdk.protocol.models.MessageKind
+
+/**
+ * Handles the submission of a close message by parsing the incoming message,
+ * validating the submission, and executing the specified callback if provided.
+ *
+ * @param call The [ApplicationCall] instance representing the HTTP call.
+ * @param exchangesApi The [ExchangesApi] instance for interacting with TBDex exchanges.
+ * @param callback The optional callback function to be executed after a successful close submission.
+ */
+@Suppress("TooGenericExceptionCaught", "MaxLineLength")
+suspend fun submitClose(
```

Comment:
Throughout the web5-kt codebase we've chosen to have exceptions be types, with the name of the exception signaling what the error was. An example of this is https://github.com/TBD54566975/web5-kt/blob/48ce6764163b32714be7d8282bfad0678f49debb/credentials/src/main/kotlin/web5/sdk/credentials/ExceptionDeclarations.kt#L10

The rationale of such approach was discussed in https://github.com/TBD54566975/web5-kt/issues/144

Is this something that makes sense to adopt in `tbdex-kt`?

If so, I'd suggest implementing as shown below, using a custom plugin in ktor:

```kt
val ExceptionsPlugin = createApplicationPlugin(name = "ExceptionsPlugin") {
  println("ExceptionsPlugin is installed!")
  on(CallFailed) { call, cause ->
    when(cause) {
      is MyCustomException -> {
        val errorDetail = ErrorDetail(detail = cause.message.orEmpty())
        val errorResponse = ErrorResponse(listOf(errorDetail))
        call.respond(HttpStatusCode.BadRequest, errorResponse)
      }
      else -> {
        println("Unknown error: ${cause.message}")
        call.respond(HttpStatusCode.BadRequest, ErrorResponse(listOf(ErrorDetail(detail = cause.message.orEmpty()))))
      }
    }
  }
}
```

The plugin can be installed in the application by doing the following within the `configure` method.

```kt
app.install(ExceptionsPlugin)
```

Personally, I'd advocate for this approach, as it decouples the translation from `Exceptions` to `ErrorResponse` types into a single place (the plugin). This makes it very easy to reuse `Exceptions`. Additionally, it ensures that the server _always_ responds with `ErrorResponse`, even in the event of what unexpected exceptions being thrown (though perhaps that's caught elsewhere in the ktor framework!).

Reply:
Oh awesome, thanks! Yeah I think we are aligning on something similar in tbdex-js too, so it would make sense to adopt here.

Reply:
Created an issue for it: https://github.com/orgs/TBD54566975/projects/29/views/1?pane=issue&itemId=48150122